src/Modal/index.tsx

```diff
@@ -88,36 +128,48 @@ const Modal: FunctionComponent<Props> = ({
     }
   }
 
+  const headerProps = {
+    variant: header.variant || 'h1',
+    align: header.align || 'left',
+    className: styles['modal-header']
+  }
+
+  const subheaderProps = {
+    variant: subheader.variant || 'h2',
+    align: subheader.align || 'left',
+    className: styles['modal-subheader']
+  }
+
+  const customModalClasses = rest.className ? rest.className + '' : ''
+  const modalClasses = clsx(styles['modal'], isShowing && styles['active'], styles[borderStyle], customModalClasses)
+  const modalWrapperClasses = clsx(styles['modal-wrapper'], isShowing && styles['active'])
+
   return (
     <Portal>
-      <div className={clsx(styles['modal-wrapper'], isShowing && styles['active'])}>
+      <div className={modalWrapperClasses}>
         <Fade duration={duration} in={isOpen}>
           <div className={styles['modal-overlay']} onClick={hideModal} onKeyDown={handleKeys} />
-            <div
-              className={clsx(styles['modal'], isShowing && styles['active'])}
-              role="dialog"
-              aria-modal="true"
-              onKeyDown={handleKeys}
-              ref={modalRef}
-              {...rest}
-            >
-              <>
-                <div
-                  role="button"
-                  tabIndex={0}
-                  onClick={hideModal}
-                  className={styles.close}
-                  aria-label="Close Modal"
-                  ref={closeBtnRef}
-                >
-                  <span aria-hidden="true">&times;</span>
-                </div>
-              </>
-              <div className="modal-content">
-                <h1>{header}</h1>
-                {children}
+          <div role="dialog" aria-modal="true" onKeyDown={handleKeys} ref={modalRef} {...rest} className={modalClasses}>
+            <>
+              <div
+                role="button"
+                tabIndex={0}
+                onClick={hideModal}
```

Comment:
this i think is a miss from my old code so my bad but for some reason the enter key isn't closing the modal when the X is focused. It should ideally just be getting the click event as teh same as mouse click so something is up there. If we can't solve it that way we can just add an case to the handle keys to hideModal() on enter.

Reply:
nevermind this is a broader bug on all things with a close button. i'll make a separate ticket so it can be addressed.